src/proxy-ui-api/frontend/tests/e2e/specs/ss-backupandrestore.js

```diff
@@ -0,0 +1,270 @@
+ * On backup & restore page delete backup with the given name and make sure it is removed
+ *
+ * @param browser
+ * @param backupFilename
+ */
+const deleteBackup = function (browser, backupFilename) {
+  const mainPage = browser.page.ssMainPage();
+  const backupAndRestoreTab =
+    mainPage.section.settingsTab.section.backupAndRestoreTab;
+  const deleteBackupConfirmationDialog =
+    backupAndRestoreTab.section.deleteBackupConfirmationDialog;
+
+  browser.waitForElementVisible(
+    `//table[contains(@class, "xrd-table")]//tr//td[text() = "${backupFilename}"]`,
+  );
+  backupAndRestoreTab.clickDeleteForBackup(backupFilename);
+  browser.waitForElementVisible(deleteBackupConfirmationDialog);
+  deleteBackupConfirmationDialog.confirm();
+  browser.waitForElementVisible(mainPage.elements.snackBarMessage);
+
+  // Make sure backup was successfully deleted
+  browser.assert.containsText(
+    mainPage.elements.snackBarMessage,
+    `Backup ${backupFilename} deleted`,
+  );
+  mainPage.closeSnackbar();
+};
+
+/**
+ * Navigate to restore and backup page when browser is started
+ *
+ * @param browser
+ */
+const navigateToBackupAndRestorePage = function (browser) {
+  const frontPage = browser.page.ssFrontPage();
+  const mainPage = browser.page.ssMainPage();
+  const settingsTab = mainPage.section.settingsTab;
+  const backupButton =
+    settingsTab.section.backupAndRestoreTab.elements.backupButton;
+
+  // Open SUT and check that page is loaded
+  frontPage.navigate();
+  browser.waitForElementVisible('//*[@id="app"]');
+
+  // Enter valid credentials
+  frontPage.signinDefaultUser();
+
+  // Navigate to settings tab
+  mainPage.openSettingsTab();
+  browser.waitForElementVisible(settingsTab);
+  settingsTab.openBackupAndRestore();
+  browser.waitForElementVisible(backupButton);
+};
+
+module.exports = {
+  tags: ['ss', 'backupandrestore'],
+  'Security server backups can be created, listed, filtered and removed': async (
+    browser,
+  ) => {
+    const mainPage = browser.page.ssMainPage();
+    const settingsTab = mainPage.section.settingsTab;
+    const backupAndRestoreTab = settingsTab.section.backupAndRestoreTab;
+    const deleteBackupConfirmationDialog =
+      backupAndRestoreTab.section.deleteBackupConfirmationDialog;
+
+    navigateToBackupAndRestorePage(browser);
+
+    // Create backup
+    backupAndRestoreTab.clickCreateBackup();
+    browser.waitForElementVisible(mainPage.elements.snackBarMessage);
+
+    // Make sure backup was successfully created
+    browser.assert.containsText(
+      mainPage.elements.snackBarMessage,
+      'successfully created',
+    );
+
+    // Get the backend-generated name of the backup and close snackbar
+    const createdBackupFileNameTextObject = await browser.getText(
+      'xpath',
+      mainPage.elements.snackBarMessage,
+    );
+    const createdBackupFileName = createdBackupFileNameTextObject.value.split(
+      ' ',
+    )[1];
+    console.log('Created backup: ', createdBackupFileName);
```

Comment:
Is this `console.log` needed?

Reply:
Not necessarily but I left it as it could prove usefull in some situation when trying to figure out why this test is failing. If it's not logged the name of the created backup will be lost. Didn't remove it yet but I don't have strong opinions about it so if you feel it should be removed then we will.

Reply:
I guess there's no need to remove it since it is used elsewhere in the tests also and it's not like it clutters any production logs anyway